src/pathstringifier.ts

```diff
@@ -1,21 +1,31 @@
 export type StringifiedPath = string;
 type Path = string[];
 
-export const escapeKey = (key: string) => key.replace(/\./g, '\\.');
+export const escapeKey = (key: string) =>
+  key.replace(/\\/g, '\\\\').replace(/\./g, '\\.');
 
 export const stringifyPath = (path: Path): StringifiedPath =>
   path
     .map(String)
     .map(escapeKey)
     .join('.');
 
-export const parsePath = (string: StringifiedPath) => {
+export const parsePath = (string: StringifiedPath, legacyPaths: boolean) => {
   const result: string[] = [];
 
   let segment = '';
   for (let i = 0; i < string.length; i++) {
     let char = string.charAt(i);
 
+    if (!legacyPaths) {
+      const isEscapedBackslash = char === '\\' && string.charAt(i + 1) === '\\';
+      if (isEscapedBackslash) {
+        segment += '\\';
+        i++;
+        continue;
+      }
+    }
```

Comment:
This correctly handles escaped backslashes, but does not implement special behavior for backslashes that are in final position (e.g., `"foo.bar.baz\\"`) or followed by something other than a dot or backslash (e.g., `"foo.bar.b\\az"` or `"foo.bar.b\\\n"`). I strongly recommend rejecting such input rather than treating it as synonymous with e.g. `"foo.bar.baz\\\\"`/`"foo.bar.b\\\\az"`/`"foo.bar.b\\\\\n"` (respectively).
```suggestion
    if (!legacyPaths && char === '\\') {
      const escaped = string.charAt(i + 1);
      if (escaped === '\\') {
        segment += '\\';
        i++;
        continue;
      } else if (escaped !== '.') {
        throw Error('invalid path');
      }
    }
```

Reply:
good point, addressed in https://github.com/flightcontrolhq/superjson/pull/311/commits/38df19518819b9689a51cd80b364da8060f3fdf2.